### ISSUE TITLE
No system has yet been discovered. Please see #77 if it doesn't resolve itself in a few seconds.

### ISSUE THREAD

**Issue by dolphinkite** [jishi/node-sonos-http-api#105]
I get the above error: "No system has yet been discovered. Please see #77 if it doesn't resolve itself in a few seconds." when I try to access http://localhost:5005/zones using curl and a terminal on the docker container, or when I try from another computer (http://mynas:5005/zones).

Looking at issue #77, this seems different since it is related to docker. 

I'm running the docker image on a Synology NAS. Synology has the option of using a Gui/Wizard to launch the images/create the containers, or to specify the Docker Run parameters. Unfortunately, the --net and --restart options do not seem available (according to the IMAGE section here: http://blog.pavelsklenar.com/how-to-install-and-use-docker-on-synology/ ). However, I also tried to start the Docker from SSH using the "docker run --net=host --restart=always -d -p 5005:5005" command and I didn't see a different behavior.

Running the current code (from git) on my Windows laptop, I am able to control my Sonos (although there are issues with TTS).

When I open a terminal on my Docker image (using Synology's Docker platform) and try something like curl http://localhost:5005/zones I get the same error described in the previous comment: "No system has yet been discovered. Please see #77 if it doesn't resolve itself in a few seconds."

This is my first experience with Docker, so my learning curve it steep, but if I can share logs or anything here, I'll be happy to do it. Can you help,  @riemers ?

How can I debug this further? Is there a way to confirm is it is a problem of multicast packets not reaching my docker? I tried to ping my sonos players from the docker terminal and it worked without a problem. 

Thanks.

**Comment by riemers** [comment]
Not at home right now, but for one i can tell you that you do indeed need to run the docker image from the command line. The gui for synology doesn't allow you to set those options. Once i did that it was working for me. I don't know how old/new your DSM is for synology.

**Comment by dolphinkite** [comment]
My DSM is *definitely* running the latest of DSM and all its packages. 

I find the synology docker app to be unstable (although this is my first and only experience using docker). At times, I couldn't create a terminal window, or it would tell me there were conflicting port forwarding issues when there really wasn't (if I fwd port 5005:5005 to a container, then later delete that container and try to do the same fwding with the new container, it could complain). I learned that stopping and starting the docker app will fix some of those issues.

Going back to starting with the command line, I find this surprising because I thought that perhaps the --restart and --host were "by default" in the Synology wizard, but I'll take your word that it is not.

Here are the command lines I am using when I ssh: 
docker run --net=host --restart=always --name Sonos-http-container -d -p 5015:5005 riemers/sonos_http_api

I have not set any env variables nor any disk mount.

At the moment I'm unable to get back there due to another issue (conflicting ports) for which I will file a separate issue (although I suspect it is a problem with Synology Docker...)

**Comment by dolphinkite** [comment]
Ok. I'm confused. It now works... :|.

I was having trouble reproducing the problem because I wasn't able to start the container anymore (was getting a conflict with port 5005). I tried to run a terminal in the container to edit the server.js to configure it on a different server port, but each time I stopped and started the container, the file got overwritten (I think it is because it pulls the files from github each time it starts. I'm no expert, but it seems a surprising practice to include those commands in dockerstart.sh since it means that at any time my container can become broken if I need to restart it...?).

Anyway, running netstat -punta | grep 5005 I was noticing a lighttpd process using port 5005 (not sure why?? webdav on my synology is off for port 5005) and as I was issuing commands to start and stop the container, my Synology experienced a HARD CRASH (!).

After everything was restarted, all of the sudden, the commands were working. I can't explain, except to say that Synology's docker is far from stable.
